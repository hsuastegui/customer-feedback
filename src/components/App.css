.app {
  max-width: 90%;
  margin: 20px auto;
  background: #FFF;
  border-radius: 5px;
  padding: 10px;
}

@media (min-width: 768px) {
  .app {
    width: 50%;
    min-width: 600px;
    max-width: 800px;
    display: grid;
    grid-template-rows: 1fr 1fr;
    grid-template-columns: 1fr 1fr;
  }
}