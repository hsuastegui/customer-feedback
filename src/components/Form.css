.form {
    display: block;
    padding: 5px;
}

.form label {
    font-size: 12px;
    margin-bottom: 2px;
    display: block;
    color: #232126;
}

.form input:not([type="range"]),
.form textarea {
    display: block;
    width: 100%;
    border-color: #232126;
    border-radius: 5px;
    border-width: 1px;
    line-height: 26px;
    margin-bottom: 10px;
    padding: 0 5px;
    box-sizing: border-box;
}

.form input[type="range"] {
    display: block;
    width: 100%;
    line-height: 26px;
    margin-bottom: 10px;
}

.form button {
    display: block;
    background: #232126;
    color: #D2D9CC;
    height: 36px;
    border-radius: 5px;
    width: 100%;
    cursor: pointer;
}

.form svg:hover {
    cursor: pointer;
    transform: scale(1.1, 1.1);
}