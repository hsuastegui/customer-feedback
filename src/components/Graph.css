.chart {
    padding: 0 10px;
}

.ct-grid {
    stroke: #0003;
    stroke-width: 1px;
    stroke-dasharray: 2px;
}

.ct-area {
    fill: #D9843B;
}

.ct-line {
    stroke: #BF5630;
    stroke-width: 2px;
    fill: none;
}

.ct-point {
    stroke: #BF5630;
    stroke-width: 10px;
    stroke-linecap: round;
}