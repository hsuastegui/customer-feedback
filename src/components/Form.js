import React, { useRef, useState } from "react";
import PropTypes from "prop-types";
import Rating from "./Rating";
import "./Form.css";

function Form({ handleSubmit }) {
  const [rating, setRating] = useState(3);
  const formRef = useRef(null);
  const clearForm = (elements) => {
    setRating(3);
    elements.forEach((e) => {
      e.value = "";
    });
  };
  const onSubmit = (e) => {
    e.preventDefault();
    const elements = formRef.current.querySelectorAll("input, textarea");
    const data = Object.values(elements).reduce(
      (acc, curr) => ({ ...acc, [curr.name]: curr.value }),
      {}
    );
    data.rating = rating;
    handleSubmit(data);
    clearForm(Object.values(elements));
  };
  return (
    <form className="form" ref={formRef} onSubmit={onSubmit}>
      <label htmlFor="name">Name</label>
      <input type="text" name="name" required />
      <label htmlFor="email">Email</label>
      <input type="email" name="email" required />
      <label>Rating</label>
      <Rating rating={rating} setRating={setRating} />
      <label htmlFor="comment">Comment</label>
      <textarea name="comment" required></textarea>
      <button type="submit">Send</button>
    </form>
  );
}

Form.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
};

export default Form;
