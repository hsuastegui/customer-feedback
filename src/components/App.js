import React, { useState } from "react";
import Form from "./Form";
import Graph from "./Graph";
import List from "./List";
import "./App.css";

const initialState = [
  { name: "John", email: "john@email.com", rating: 3, comment: "Lorem ipsum" },
  { name: "Jane", email: "jane@email.com", rating: 2, comment: "Lorem ipsum" },
];

function App() {
  const [comments, setComments] = useState(initialState);

  const handleSubmit = (data) => {
    setComments([...comments, data]);
  };

  return (
    <div className="app">
      <Form handleSubmit={handleSubmit} />
      <Graph data={comments} />
      <List data={comments} />
    </div>
  );
}

export default App;
