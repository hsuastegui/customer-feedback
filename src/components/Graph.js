import React, { useEffect } from "react";
import PropTypes from "prop-types";
import Chartist from "chartist";
import "./Graph.css";

function Graph({ data }) {
  const lineData = data.map(({ rating }) => rating);
  const labels = data.map((d, i) => i + 1);
  useEffect(() => {
    new Chartist.Line(
      "#chart",
      {
        labels,
        series: [lineData],
      },
      {
        showPoint: true,
        lineSmooth: true,
        low: 0,
        high: 5,
        showArea: true,
      }
    );
  }, [labels, lineData]);
  return <div className="chart" id="chart"></div>;
}

Graph.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      email: PropTypes.string,
      rating: PropTypes.number,
      comment: PropTypes.string,
    }).isRequired
  ),
};

export default Graph;
