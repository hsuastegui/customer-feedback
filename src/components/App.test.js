import React from "react";
import { shallow } from "enzyme";
import App from "./App";
import Form from "./Form";
import Graph from "./Graph";

describe("App", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });
  it("renders app", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("handles submit", () => {
    const item = {
      name: "Doe",
      email: "doe@email.com",
      rating: 5,
      comment: "Consectetur adipiscing elit",
    };
    const handleSubmit = wrapper.find(Form).prop("handleSubmit");
    handleSubmit(item);
    expect(wrapper.find(Graph).prop("data")[2]).toEqual(item);
  });
});
