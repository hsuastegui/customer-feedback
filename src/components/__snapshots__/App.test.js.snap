// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders app 1`] = `
<div
  className="app"
>
  <Form
    handleSubmit={[Function]}
  />
  <Graph
    data={
      Array [
        Object {
          "comment": "Lorem ipsum",
          "email": "john@email.com",
          "name": "John",
          "rating": 3,
        },
        Object {
          "comment": "Lorem ipsum",
          "email": "jane@email.com",
          "name": "Jane",
          "rating": 2,
        },
      ]
    }
  />
  <List
    data={
      Array [
        Object {
          "comment": "Lorem ipsum",
          "email": "john@email.com",
          "name": "John",
          "rating": 3,
        },
        Object {
          "comment": "Lorem ipsum",
          "email": "jane@email.com",
          "name": "Jane",
          "rating": 2,
        },
      ]
    }
  />
</div>
`;
